plugins/woocommerce-admin/client/marketplace/components/feedback-modal/feedback-modal.tsx

```diff
@@ -0,0 +1,244 @@
+	const openModal = () => setOpen( true );
+	const closeModal = () => {
+		incrementDismissedTimes();
+		setOpen( false );
+	};
+	const { createNotice } = useDispatch( 'core/notices' );
+
+	function maybeShowSnackbar() {
+		// don't show if the user has already given feedback or otherwise suppressed:
+		if ( isDismissedForever() ) {
+			return;
+		}
+
+		// don't show if the user has already declined to provide feedback today:
+		const today = new Date().toDateString();
+		if (
+			today ===
+			localStorage.getItem( LOCALSTORAGE_KEY_LAST_REQUESTED_DATE )
+		) {
+			return;
+		}
+
+		createNotice(
+			'success',
+			__( 'How easy is it to find an extension?', 'woocommerce' ),
+			{
+				type: 'snackbar',
+				icon: (
+					<>
+						<svg
+							color="#fff"
+							strokeWidth="1.5"
+							viewBox="0 0 28.873 8.9823"
+							style={ { height: '8px', marginLeft: '-7px' } }
+						>
+							<path
+								className="l"
+								d="m4.1223 1.1216 19.12-0.014142 4.3982 3.38-4.3982 3.38-19.12-0.014142a3.34 3.34 0 0 1-2.39-0.97581 3.37 3.37 0 0 1 0.00707-4.773 3.34 3.34 0 0 1 2.383-0.98288z"
+								stroke="#fff"
+							/>
+							<line
+								className="l"
+								x1="6.7669"
+								x2="6.7669"
+								y1="7.8533"
+								y2="1.1216"
+								stroke="#fff"
+							/>
+							<path
+								className="l"
+								d="m23.235 1.1146 4.4053 3.3729-4.3982 3.38a6.59 6.59 0 0 1-0.89096-3.3517 6.59 6.59 0 0 1 0.88388-3.4012z"
+								stroke="#fff"
+							/>
+							<line
+								className="l"
+								x1="6.7669"
+								x2="22.323"
+								y1="4.4875"
+								y2="4.4875"
+								stroke="#fff"
+							/>
+						</svg>
+					</>
+				),
+				explicitDismiss: true,
+				onDismiss: incrementDismissedTimes,
+				actions: [
+					{
+						onClick: openModal,
+						label: 'Give feedback',
+					},
+				],
+			}
+		);
+	}
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps -- [] => we only want this effect to run once, on first render
+	useEffect( maybeShowSnackbar, [] );
+
+	// We don't want the "How easy was it to find an extension?" dialog to appear forever:
+	const FEEDBACK_DIALOG_CAN_APPEAR =
+		new Date( SUPPRESS_IF_AFTER_DATE ) > new Date();
+	if ( ! FEEDBACK_DIALOG_CAN_APPEAR ) {
+		return <></>;
+	}
+
+	function easyToFindChanged( value: number ) {
+		setEasyToFindValidiationFailed( false );
+		setEasyToFind( value );
+	}
+
+	function meetsMyNeedsChanged( value: number ) {
+		setMeetsMyNeedsValidiationFailed( false );
+		setMeetsMyNeeds( value );
+	}
+
+	function submit() {
+		// Validate:
+		if ( easyToFind === 0 || meetsMyNeeds === 0 ) {
+			if ( easyToFind === 0 ) setEasyToFindValidiationFailed( true );
+			if ( meetsMyNeeds === 0 ) setMeetsMyNeedsValidiationFailed( true );
+			return;
+		}
+		// Send event to CES:
+		recordEvent( 'ces_feedback', {
+			action: CUSTOMER_EFFORT_SCORE_ACTION,
+			score: easyToFind,
+			score_second_question: meetsMyNeeds,
+			score_combined: easyToFind + meetsMyNeeds,
+			comments: thoughts,
```

Comment:
I think we should use `wp.escapeHtml` for comments to sanitize user input. Let me know if we're handling it somewhere else.

Reply:
Good point! It's not good to use a global, so instead, I've used the existing `sanitizeHTML` function in [`216e306` (#39609)](https://github.com/woocommerce/woocommerce/pull/39609/commits/216e3061787923e070878c1491098fb984f53f47).